src/trace/trace.cpp

```diff
@@ -637,25 +637,26 @@ void Trace::merge_ips(IpRefMap& new_children, IpCctxMap& children,
     }
 }
 
-otf2::definition::mapping_table Trace::merge_ips(IpRefMap& new_ips,
-                                                 std::map<pid_t, ProcessInfo>& infos)
+otf2::definition::mapping_table Trace::merge_calling_contexts(
+    IpRefMap& new_ips,
+    const std::unordered_map<pid_t, otf2::definition::calling_context::reference_type>& thread_refs,
+    std::map<pid_t, ProcessInfo>& infos)
```

Comment:
I think there is something wrong here outside of this snippet. This reference should be `const`; we're only reading from it. What we should change is, that the mutex in `ProcessInfo` should be `mutable` and therefore, `ProcessInfo::maps()` can be `const`. Don't you believe me? But you will believe Herb Sutter: https://channel9.msdn.com/posts/C-and-Beyond-2012-Herb-Sutter-You-dont-know-blank-and-blank

This might be done as another PR though.

Reply:
also `new_ips` should be `const`